### PR TITLE
FIT-6 add new functional tests

### PR DESCRIPTION
**Pull request by tparkerosi**
Added functional testing + tests.

## Description
Added RestAssure functional testing as part of build and added two tests.

## Jira Issue link
https://osi-cwds.atlassian.net/browse/FIT-6

## Tests
- [ ] I have included unit tests 
- [x] I have included functional tests -Rest based End to End tests
- [ ] I have included other tests 
- [x] I have NOT included tests 
I have not included unit tests for test code. This code is only used during functional testing phase and is tested from within tests. 

## Types of changes

- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the google code style of this project.
- [x] I have ran all tests for the project.
- [x] My code is in a stable state ready to be deployable, but not necessarily complete.
- [x] I promise on my honor as a CWDS developer to ensure I don't break the build, to check SonarQube after the build completes, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

File: config/testConfig.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+authenticationMode: TEST
+testUrl:
+  baseUrl: ${BASE_URL}
+  authenticationBaseUrl: ${BASE_URL}
```
Comment (AVandana-OSI) [inline config/testConfig.yml]: I see both baseUrl and autheincationBaseUrl in the yml file, but we discussed it should be autheincationBaseUrl only. If I am wrong please ignore.

Comment (tparkerosi) [reply config/testConfig.yml]: Base Url is the url of the application we are testing. Authentication URL is the url for perry.

Comment (AVandana-OSI) [reply config/testConfig.yml]: @tparkerosi  Can we use call back URL as the same application we are testing, Or else we can change the baseUrl to applicationTestUrl or applicationBaseUrl.

Comment (tparkerosi) [reply config/testConfig.yml]: call back url is from the point of view of the auth app. It wouldn't make sense in the rest of the tests. The base url is the base url for the application. We could change it to applicationUrl, but I would prefer something shorter. applicationTestUrl doesn't really make sense. The Url is for the Application end point, not the test. We could use applicationBaseUrl, but not sure why we would want to qualify it. Could you elaborate on why?

Comment (AVandana-OSI) [reply config/testConfig.yml]: I was thinking maybe base url might get confused with the perry url, but applicationUrl sounds good to me.

**Comment by tparkerosi** [comment]
retest this please

**Pull request merged**